### ISSUE TITLE
[Question] Any DSDT\kext patches to fully disable Intel WiFi&Bluetooth?

### ISSUE THREAD

**Issue by ghost** [daliansky/XiaoMi-Pro-Hackintosh#13]
Hello there! I'm still waiting my notebook, but even now dreaming about hackintosh setup. My question is - how can i fully disable internal Intel 8265? I want to install DW1560 or DW1830 with [adapter](https://www.allekabels.nl/m2-ssd/15535/1445761/m2-ngff-adapter.html) (didn't know exactly which, because i don't know which connectors and how much of them on original antenas), but i guess it will conflict with existing Intel BT hardware. Also - which patches i need to apply to get one of this cards (advice which is better please) working?

**Comment by kvn1351** [comment]
You'll have to disable it in by removing the definition for the usb port responsible for the bluetooth module within the injector. To find the correct port, check your IOReg.

**Comment by PavelLJ** [comment]
Thanks for reply! Okay, i understood that part. What about new WiFi card? What i will need to do to make this thing work? Even on physical side - i'm not sure that our *Tri-Band* wifi have 3 antennas which i can connect to DW1830. I know it's not really related to this topic, but it would be very helpful for me if you will answer this questions.

**Comment by stevezhengshiqi** [comment]
It seems like Intel WiFI is consuming power though it is not working. From the IORegExplorer, the RP08.PXSX device(Intel Wireless 8265 for Xiaomi-Pro)'s `CurrentPowerState` is `0x2`, and we expect `0x0` to fully disable it. Actually, the `PXSX` device of RP08 should not exist, just like RP01(Discrete Graphics Card).
![Screen Shot 2019-05-22 at 18 45 35](https://user-images.githubusercontent.com/36894554/58169728-0b81c900-7cc4-11e9-9737-4f76d70b9f41.jpg)

The first method is to call `_OFF` method in DSDT. However, the `_OFF` method for `RP08.PXSX` is empty, and it's too hard for me to complete.

The second method is to patch through kernel extensions. For example, in `WhateverGreen` project, https://github.com/acidanthera/WhateverGreen/commit/b2e7cd4c3147088093af7f1992a0fd7395d623ef disables all external GPUs in I/O side.

Both methods take time, and I may not capable of finishing my thoughts. Hope someday Intel Wifi macOS driver is released, then we won't have this problem.

**Comment by Menchen** [comment]
There's a third method, disable internal wifi in bios, the downside I think is that you won't be able to use intel wifi in windows(dual boot). But it's fine if you got an extra wifi card in the m.2 slot...

**Comment by stevezhengshiqi** [comment]
@Menchen Hi, based on daliansky's [OC-liittle](https://github.com/daliansky/OC-little/tree/master/22-禁止PCI设备), I made an SSDT designed for disabling the internal WiFi. Could you help me test it?
[SSDT-DRP08.aml.zip](https://github.com/daliansky/XiaoMi-Pro-Hackintosh/files/3954230/SSDT-DRP08.aml.zip)

**Comment by Menchen** [comment]
@stevezhengshiqi I got an exam tomorrow, but I can test it at the weekend.

**Comment by Menchen** [comment]
@stevezhengshiqi After installing that SSDT and rebooting, my intel Bluetooth still works, so It seems still using power..... 

But the weird thing is I got 4.5~5 hours of battery life estimation! (Still need more real-life test), almost 1.5 hours more without the patch! (I just used yesterday and battery life estimation was 3 hours and 30minute.)

I have the following settings:
- MSR 0xe2 unlock and HWP enabled(with some tune)(20~30m battery gain, but much better thermal)
-Disabled EGPU https://github.com/daliansky/XiaoMi-Pro-Hackintosh/issues/224#issuecomment-559897974(almost an hour gain)
- TDP change and CPU Undervolting (-110mah, far better thermal and battery life)(half-hour gain I think)
- SSDT-DRP08 patch you just posted. (almost 1.5 hours gain)

About 10~12W saved with all the above optimization.

So in conclusion, I think intel Bluetooth must have a separated power line(maybe by USB) or the SSDT disabled wifi only and not the entire Wifi module (I think in Linux it's possible). But in either case, this patch is awesome.

Thank you for this great patch!

**Comment by stevezhengshiqi** [comment]
@Menchen Happy to hear that! Could you post an ioreg file of your device after patching SSDT-DRP08? We can see more detail from the ioreg file.

**Comment by Menchen** [comment]
@stevezhengshiqi  
[AFTERPATCH.ioreg.zip](https://github.com/daliansky/XiaoMi-Pro-Hackintosh/files/3962415/AFTERPATCH.ioreg.zip)

**Comment by stevezhengshiqi** [comment]
@Menchen From your ioreg, the PXSX device of RP08 is gone, which may represent that the wireless device is disabled. I will post the patch to the repository.

**Comment by buliaoyin** [comment]
@stevezhengshiqi Tried this patch, seems no effect on GTX version

**Comment by stevezhengshiqi** [comment]
@buliaoyin Please attach the ioreg file for closer look.

**Comment by buliaoyin** [comment]
@stevezhengshiqi The PXSX device of RP08 was gone after patch (shown as below screenshot), but the power consumption is exactly the same as before.
![image](https://user-images.githubusercontent.com/4070520/70850452-6891d600-1ec5-11ea-9daf-4240262381f2.png)

**Comment by stevezhengshiqi** [comment]
That’s nice if PXSX device is gone. If you check RP01, which is the discrete graphic card, you will find that it has the same power state as that of RP08. I will add this patch by the way, and it should bring better power consumption.
